### DISCUSSION TITLE
On demand revalidation with basic authentication

### DISCUSSION THREAD

**Discussion by SHoogland** [vercel/next.js#38412] [Ideas]
I threw together some quick solution to enable the use of revalidation behind basic authentication (our test environment)

It would be awesome to get this integrated

The working code: 
https://github.com/vercel/next.js/compare/canary...SHoogland:next.js:patch-1

And a altered example to test its workings:
https://github.com/vercel/next.js/compare/canary...SHoogland:next.js:revalidate-test

Feedback appreciated!

**Comment by martijnvanonz** [comment]
This would be great :)

**Comment by lauraspina** [comment]
great feature!
